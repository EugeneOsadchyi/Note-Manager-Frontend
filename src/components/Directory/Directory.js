import React from 'react';
import PropTypes from 'prop-types';

const Directoty = ({ name }) => (
  <div>
    <span className="name">
      {name}
    </span>
  </div>
);

Directoty.propTypes = {
  name: PropTypes.string.isRequired,
};

export default Directoty;
