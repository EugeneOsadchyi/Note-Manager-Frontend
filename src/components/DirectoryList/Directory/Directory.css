.Directory.active {
  background-color: blue;
}
