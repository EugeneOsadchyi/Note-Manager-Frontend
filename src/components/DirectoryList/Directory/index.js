import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFolder, faFolderOpen } from '@fortawesome/free-solid-svg-icons';


const Directory = ({
  name,
  opened,
  folderClicked,
  doubleClicked,
}) => {
  const icon = opened ? faFolderOpen : faFolder;

  return (
    <div className="Directory" onDoubleClick={doubleClicked}>
      <span className="icon" onClick={folderClicked}>
        <FontAwesomeIcon icon={icon} />
      </span>
      <span className="name">
        {name}
      </span>
    </div>
  );
};

Directory.propTypes = {
  name: PropTypes.string.isRequired,
  opened: PropTypes.bool.isRequired,
  doubleClicked: PropTypes.func.isRequired,
  folderClicked: PropTypes.func.isRequired,
};

export default Directory;
