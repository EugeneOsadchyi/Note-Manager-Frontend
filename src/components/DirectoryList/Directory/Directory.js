import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faFolder, faFolderOpen } from '@fortawesome/free-solid-svg-icons';
import './Directory.css';

const Directory = ({
  name,
  opened,
  active,
  clicked,
  doubleClicked,
  folderClicked,
}) => {
  const icon = opened ? faFolderOpen : faFolder;
  const classes = classNames('Directory', {
    active,
  });

  return (
    <div
      className={classes}
      onClick={clicked}
      onDoubleClick={doubleClicked}
      role="presentation"
    >
      <span className="icon" role="presentation" onClick={folderClicked}>
        <FontAwesomeIcon icon={icon} />
      </span>
      <span className="name">
        {name}
      </span>
    </div>
  );
};

Directory.propTypes = {
  name: PropTypes.string.isRequired,
  opened: PropTypes.bool.isRequired,
  active: PropTypes.bool.isRequired,
  clicked: PropTypes.func.isRequired,
  doubleClicked: PropTypes.func.isRequired,
  folderClicked: PropTypes.func.isRequired,
};

export default Directory;
