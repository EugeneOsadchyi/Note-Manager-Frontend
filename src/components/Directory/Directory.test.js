import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Directory from './Directory';

describe('Directory', () => {
  const name = 'Test Directory';
  const props = { name };
  const directory = shallow(<Directory {...props} />);

  it('renders properly', () => {
    expect(shallowToJson(directory)).toMatchSnapshot();
  });

  it('displays directory name from the props', () => {
    expect(directory.find('.name').text()).toEqual(name);
  });
});
