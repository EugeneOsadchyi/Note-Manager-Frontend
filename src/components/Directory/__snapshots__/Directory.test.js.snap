// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory renders properly 1`] = `
<div>
  <span
    className="name"
  >
    Test Directory
  </span>
</div>
`;
